From a1b2c3d4e5f6a1b2c3d4e5f6a1b2c3d4e5f6a1b2 Mon Sep 17 00:00:00 2001
From: Ajender Reddy Mallikarjuna <ajenderr@qti.qualcomm.com>
Date: Fri, 07 Aug 2026 08:36:32 +0000
Subject: [PATCH 1/4] ksync: add #include <linux/vmalloc.h>

vzalloc() is used in ksync_zalloc() but <linux/vmalloc.h> is not
included. Newer kernels no longer pull it in transitively.

Upstream-Status: Inappropriate [kernel 7.x: vzalloc needs explicit vmalloc.h]
Signed-off-by: Ajender Reddy Mallikarjuna <ajenderr@qti.qualcomm.com>
---
 include/ksync.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/include/ksync.h b/include/ksync.h
index 0000000..0000001 100644
--- a/include/ksync.h
+++ b/include/ksync.h
@@ -14,6 +14,7 @@
 #include <linux/spinlock.h>
 #include <linux/sync_file.h>
 #include <uapi/linux/msm_ksync.h>
+#include <linux/vmalloc.h>
 #include <linux/file.h>
 #include <linux/list.h>
